frontend/tests/global-setup.mjs

```diff
@@ -0,0 +1,493 @@
+        console.log(`  Still waiting... (attempt ${i + 1}/${maxAttempts})`);
+      }
+    } catch (_error) {
+      // Port not ready yet
+      if ((i + 1) % 10 === 0) {
+        console.log(`  Still waiting... (attempt ${i + 1}/${maxAttempts})`);
+      }
+    }
+    await new Promise((resolve) => setTimeout(resolve, delayMs));
+  }
+  throw new Error(`Port ${port} failed to become available after ${maxAttempts} attempts`);
+}
+
+async function waitForFrontend(maxAttempts = 60, delayMs = 1000) {
+  console.log('Waiting for frontend to compile and serve...');
+
+  let consecutiveFastResponses = 0;
+  const requiredConsecutiveFast = 3;
+
+  for (let i = 0; i < maxAttempts; i++) {
+    const requestStart = Date.now();
+    try {
+      const { stdout } = await execAsync('curl -s http://localhost:3000/');
+
+      // Check for essential React app elements
+      const hasRootDiv = stdout.includes('id="root"') || stdout.includes("id='root'");
+      const hasScriptTag = stdout.includes('<script');
+      const isHTML = stdout.includes('<!doctype html') || stdout.includes('<!DOCTYPE html') || stdout.includes('<html');
+      const requestTime = Date.now() - requestStart;
+
+      if (hasRootDiv && hasScriptTag && isHTML && requestTime < 1000) {
+        consecutiveFastResponses++;
+        console.log(`  Fast response ${consecutiveFastResponses}/${requiredConsecutiveFast} (${requestTime}ms)`);
+
+        if (consecutiveFastResponses >= requiredConsecutiveFast) {
+          console.log(`✓ Frontend ready and stable (${requiredConsecutiveFast} fast responses)`);
+          return true;
+        }
+      } else {
+        if (consecutiveFastResponses > 0) {
+          console.log(`  Slow response (${requestTime}ms), resetting counter...`);
+        }
+        consecutiveFastResponses = 0;
+      }
+
+      if (requestTime > 2000) {
+        console.log(`  Attempt ${i + 1}: Slow response (${requestTime}ms), compilation in progress...`);
+      }
+    } catch (_error) {
+      consecutiveFastResponses = 0;
+    }
+    await new Promise((resolve) => setTimeout(resolve, delayMs));
+  }
+  throw new Error('Frontend failed to become ready');
+}
+
+async function setupDockerNetwork(state) {
+  console.log('Creating Docker network...');
+  const network = await new Network().start();
+  state.networkId = network.getId();
+  state.network = network;
+  console.log(`✓ Network created: ${state.networkId}`);
+  return network;
+}
+
+async function startRedpandaContainer(network, state) {
+  console.log('Starting Redpanda container...');
+  const redpanda = await new GenericContainer('redpandadata/redpanda:v25.2.1')
+    .withNetwork(network)
+    .withNetworkAliases('redpanda')
+    .withExposedPorts(
+      { container: 19_092, host: 19_092 },
+      { container: 18_081, host: 18_081 },
+      { container: 18_082, host: 18_082 },
+      { container: 9644, host: 19_644 }
+    )
+    .withCommand([
+      'redpanda',
+      'start',
+      '--smp',
```

Comment:
There is a flag: `--mode dev-container` which should speed up the start process and was made for this particular case. If we go with the Redpanda testcontainer module, this is already in there (https://github.com/testcontainers/testcontainers-node/blob/main/packages/modules/redpanda/src/redpanda-container.ts#L60)